src/aspire/operators/blk_diag_matrix.py

```diff
@@ -941,3 +946,45 @@ def diag(self):
             diag.extend(list(np.diag(blk)))
 
         return DiagMatrix(np.array(diag, dtype=self.dtype))
+
+    @staticmethod
+    def from_dense(A, blk_partition, warn_eps=None):
```

Comment:
Might be better to have `warn_eps` default to some medium value (say `1e-3`). Then users can disable the warning by setting it to `None`.

Reply:
Updated `warn_eps` default, docstring, and the line where it was being used in `SteerableBasis2D.filter_to_basis_mat`.

Reply:
Great!